### PR TITLE
Add section about database migration hook

### LINKED ISSUE #44: Add Section on how to Implement DB Migration Hooks
See also https://icinga.com/blog/2024/07/31/simplified-database-schema-upgrades-for-icinga-web-and-modules/

### PR DESCRIPTION
**Pull request by martialblog**
Fixes #44

### CODE REVIEW

**Comment by martialblog** [comment]
@nilmerg @yhabteab Could someone please confirm this is all correct? Thanks!

File: doc/48-Database-Migration-Hook.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+use ipl\Orm\Behavior\BoolCast;
+use ipl\Orm\Behavior\MillisecondTimestamp;
+use ipl\Orm\Behaviors;
+use ipl\Orm\Model;
+
+/**
+ * A database model for Training schema version table
+ *
+ * @property int $id Unique identifier of the database schema entries
+ * @property string $version The current schema version of Icinga Web
```
Comment (yhabteab) [inline doc/48-Database-Migration-Hook.md]: of ~Icinga Web~ this/the module

Comment (yhabteab) [reply doc/48-Database-Migration-Hook.md]: Otherwise, LFTM!

Comment (martialblog) [reply doc/48-Database-Migration-Hook.md]: copy-paste  https://github.com/Icinga/icingaweb2-module-reporting/blob/main/library/Reporting/Model/Schema.php#L17

Comment (martialblog) [reply doc/48-Database-Migration-Hook.md]:
```suggestion
 * @property string $version The current schema version of the module
```

**Pull request merged**